core/src/ModelComponent.cpp

```diff
@@ -0,0 +1,79 @@
+/*!
+ * @file ModelComponent.cpp
+ *
+ * @date Feb 28, 2022
+ * @author Tim Spain <timothy.spain@nersc.no>
+ */
+
+#include "include/ModelComponent.hpp"
+
+namespace Nextsim {
+
+std::map<std::string, ModelComponent*> ModelComponent::registeredModules;
```

Comment:
I suggest to use unordered_map if ordering is not required for correctness, as it has better time complexities for most operations.

Reply:
Same with set and unordered_set